react/components/FilterOptions/FilterColapsible.js

```diff
@@ -0,0 +1,112 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import merge from 'lodash/merge'
+
+import Statement from '../UncoupledStatement/UncoupledStatement'
+import Collapsible from '../Collapsible'
+
+const emptyVirtualStatement = {
+  subject: null,
+  verb: null,
+  object: null,
+  error: null,
+}
+
+const filterStatementBySubject = (statements = [], subject, options = {}) => {
+  const hasStatement = statements.some(st => st.subject === subject)
+  return hasStatement
+    ? statements.filter(st => st.subject === subject)
+    : [
+        {
+          ...emptyVirtualStatement,
+          subject: subject,
+          verb:
+            options[subject] && options[subject].verbs.length > 0
+              ? options[subject].verbs[0].value
+              : null,
+        },
+      ]
+}
+
+class FilterColapsible extends PureComponent {
+  constructor(props) {
+    super(props)
+    this.filterMenuContainer = React.createRef()
+
+    this.state = {
+      isCollapsibleOpen: false,
+      virtualStatement: filterStatementBySubject(
+        props.statements,
+        props.subject,
+        props.options
+      )[0],
+    }
+  }
+
+  handleChangeStatement = (newValue, structure) => {
+    this.setState(state => {
+      const virtualStatement = merge({}, state.virtualStatement, {
+        [structure]: newValue,
+      })
+      this.props.onChangeStatement(virtualStatement)
```

Comment:
The component is trying to sync its state with the props. Choose one. Usually we go by the props.

https://reactjs.org/docs/forms.html#controlled-components